tests/integration/test_events.py

```diff
@@ -0,0 +1,132 @@
+from hexkit.providers.akafka.testutils import KafkaFixture
+
+from sms.inject import prepare_rest_app
+from tests.fixtures.config import get_config
+from tests.fixtures.utils import VALID_BEARER_TOKEN
+
+pytestmark = pytest.mark.asyncio()
+
+TEST_EVENT1 = {
+    "type_": "test_event",
+    "payload": {"test": "event1"},
+    "key": "key1",
+    "topic": "topic1",
+}
+TEST_EVENT2 = {
+    "type_": "test_event",
+    "payload": {"test": "event2"},
+    "key": "key1",
+    "topic": "topic1",
+}
+TEST_EVENT3 = {
+    "type_": "test_event",
+    "payload": {"test": "event3"},
+    "key": "key1",
+    "topic": "topic2",
+}
+# 2 events in topic1, 1 event in topic2
+TEST_EVENTS = [TEST_EVENT1, TEST_EVENT2, TEST_EVENT3]
+TEST_TOPICS = ["topic1", "topic2"]
+
+
+@pytest.mark.parametrize(
+    "events_to_publish", [TEST_EVENTS, []], ids=["EventsPublished", "NoEventsPublished"]
+)
+@pytest.mark.parametrize(
+    "topics_to_clear",
+    [
+        [],
+        TEST_TOPICS,
+        ["topic1"],
+        ["topic2"],
+        ["does-not-exist"],
+    ],
+    ids=[
+        "ClearNoTopics",
+        "ClearAllTopics",
+        "ClearTopic1",
+        "ClearTopic2",
+        "ClearNonExistentTopic",
+    ],
+)
+async def test_clear_topics_happy(
+    kafka: KafkaFixture,
+    events_to_publish: list[dict[str, Any]],
+    topics_to_clear: list[str],
+):
+    """Test that topics can be cleared."""
+    config = get_config(sources=[kafka.config])
+
+    # Publish events if applicable
+    published_topics: set[str] = set()
+    for event in events_to_publish:
+        published_topics.add(event["topic"])
+        await kafka.publish_event(**event)
+
+    # Call the endpoint to delete the topics in topics_to_clear
+    async with (
+        prepare_rest_app(config=config) as app,
+        AsyncTestClient(app=app) as client,
+    ):
+        response = await client.delete(
+            "/events/",
+            params={"topic": topics_to_clear},
+            headers={"Authorization": VALID_BEARER_TOKEN},
+        )
+        assert response.status_code == 204
+
+    # Check that the topics have been cleared
+    if published_topics:
+        consumer = AIOKafkaConsumer(
+            *published_topics,
+            bootstrap_servers=kafka.kafka_servers[0],
+            group_id="sms",
+            auto_offset_reset="earliest",
+            enable_auto_commit=True,
+            consumer_timeout_ms=2000,
+        )
+        await consumer.start()
+
+        # Calculate how many records *should* be left
+        num_records_remaining = (
+            len(events_to_publish)
+            - len(
+                [
+                    event
+                    for event in events_to_publish
+                    if event["topic"] in topics_to_clear
+                ]
+            )
+            if topics_to_clear
+            else 0
+        )
+
+        # Verify that the topics have been cleared
+        prefetched = await consumer.getmany(timeout_ms=500)
```

Comment:
If you check `EventsPublished-ClearNoTopics` this does not seem to fetch anything and the issue is currently masked by the ternary issue above.
Might need to pass the partitions explicitly.

Reply:
The test ID is misleading (which has since been fixed). The fetch doesn't return anything because everything has been deleted. The test actually represents the case where the supplied `topics` list is empty, which means to delete all topics.